src/main/java/uk/gov/hmcts/darts/audio/helper/UnstructuredDataHelper.java

```diff
@@ -0,0 +1,146 @@
+import uk.gov.hmcts.darts.common.entity.AnnotationDocumentEntity;
+import uk.gov.hmcts.darts.common.entity.CaseDocumentEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalLocationTypeEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectRecordStatusEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.common.enums.SystemUsersEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
+import uk.gov.hmcts.darts.common.repository.UserAccountRepository;
+import uk.gov.hmcts.darts.datamanagement.api.DataManagementApi;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum.UNSTRUCTURED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+import static uk.gov.hmcts.darts.datamanagement.DataManagementConstants.MetaDataNames.MEDIA_REQUEST_ID;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class UnstructuredDataHelper {
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final DataManagementApi dataManagementApi;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final UserAccountRepository userAccountRepository;
+
+    private static final List<CompletableFuture> jobsList = new ArrayList<>();
+
+    public static List<CompletableFuture> getJobsList() {
+        return jobsList;
+    }
+
+    public boolean createUnstructured(ExternalObjectDirectoryEntity eodEntityToDelete, ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        boolean returnVal = true;
+        UUID uuid = saveToUnstructuredDataStore(eodEntity, binaryData);
+        if (uuid == null) {
+            returnVal = false;
+        } else {
+            saveToDatabase(eodEntity, uuid);
+            removeFromDatabase(eodEntityToDelete);
+        }
+        return returnVal;
+    }
+
+    private UUID saveToUnstructuredDataStore(ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        UUID uuid = null;
+        try {
+            Map<String, String> metadata = new HashMap<>();
+            metadata.put(MEDIA_REQUEST_ID, String.valueOf(eodEntity.getMedia().getId()));
+            BlobClient blobClient = dataManagementApi.saveBlobDataToContainer(binaryData, DatastoreContainerType.UNSTRUCTURED, metadata);
```

Comment:
@chrisbellingham-hmcts I forget. Did you put a pattern in to avoid use of BinaryData when uploading. Should we try to avoid its use?

Reply:
Yep, the downside of the approach here is the underlying `public BlobClient saveBlobData(String containerName, BinaryData binaryData, Map<String, String> metadata)` method in `DataManagementApiImpl` will read all data into memory.

Ideally we want to use the `public UUID saveBlobData(String containerName, InputStream inputStream)` method in `DataManagementApiImpl`, which expects an inputstream.

Reply:
@chrisbellingham-hmcts / @mestebanez - having a bit of fun with this one :-/
the job thats doing the blob saving is running in CompletableFuture. unfortunately when I switch to use the input stream, the temp input stream has been deleted before the async job gets run.
can't see any way round this apart from switching back to using BinaryData - open to suggestions :-D

Reply:
Maybe we put in a pin in this one, and look to address this via an improvement ticket. Practically, memory efficiency is less critical in ATS MODE, as we'll have a single pod dedicated to servicing the task. That being said I think it still needs solved to allow us to cater for particularly large files.

@hemantasharma1129, what's your view on this one, shall this be merged as-is or do we need this to suport large files (files whose size exceeds the memory allocation of a pod) immediately?

Reply:
I agree with @chrisbellingham-hmcts . We are suppose to use streams only as loading a file in memory is something we are no longer doing.

Reply:
sorted - as per discussion